test/test_pebble.py

```diff
@@ -3893,3 +3960,54 @@ def recv():
         test_websocket_recv_raises = pytest.mark.filterwarnings(
             'ignore::pytest.PytestUnhandledThreadExceptionWarning'
         )(test_websocket_recv_raises)
+
+
+class TestIdentity:
+    def test_local_identity_from_dict(self):
+        identity = pebble.Identity.from_dict({'access': 'admin', 'local': {'user-id': 42}})
+        assert identity == pebble.Identity(
+            access=pebble.IdentityAccess.ADMIN, local=pebble.LocalIdentity(user_id=42)
+        )
+
+    def test_local_identity_to_dict(self):
+        identity = pebble.Identity(
+            access=pebble.IdentityAccess.ADMIN, local=pebble.LocalIdentity(user_id=42)
+        )
+        assert identity.to_dict() == {'access': 'admin', 'local': {'user-id': 42}}
+
+    def test_basic_identity_from_dict(self):
+        identity = pebble.Identity.from_dict({
+            'access': 'metrics',
+            'basic': {'password': 'hashed password'},
+        })
+        assert identity == pebble.Identity(
+            access=pebble.IdentityAccess.METRICS,
+            basic=pebble.BasicIdentity(password='hashed password'),
+        )
+
+    def test_basic_identity_to_dict(self):
+        identity = pebble.Identity(
+            access=pebble.IdentityAccess.METRICS,
+            basic=pebble.BasicIdentity(password='hashed password'),
+        )
+        assert identity.to_dict() == {
+            'access': 'metrics',
+            'basic': {'password': 'hashed password'},
+        }
+
+    def test_no_access(self):
+        with pytest.raises(ValueError) as excinfo:
+            raw: pebble.IdentityDict = {'local': {'user-id': 42}}
+            pebble.Identity.from_dict(raw)
+        assert str(excinfo.value) == '"access" key is required in IdentityDict'
```

Comment:
Let's not test the exception message, just that `ValueError` is raised. Similar below.